Use new Task-oriented method of handling sign-in

i077 opened the pull request:
`handleSignInResult()` in `SignInActivity` now accepts a `Task` object rather than a `SignInResult` object.
Probably due to Google updating their APIs or something.

**We should verify this is working on some machine other than my laptop before merging.**

i077 committed:
Attempt to sign in with task

Signing in only returns an ApiException that we catch, with code
12500.

yagmur2 reviewed (commented):
Works only some of the time. Signing in with Google either crashes application, succeeds, or returns to SignInActivity screen.

i077 committed:
Add temporary snackbar for guest sign-in

Guest sign-in is not fully implemented yet, so alert user when they
try to sign in as guest.
